src/io.rs

```diff
@@ -50,3 +68,19 @@ pub fn read_embeddings_view(
 
     Ok(embeds?)
 }
+
+pub fn read_quantized_embeddings(
```

Comment:
This function currently reads any embeddings. So, it will 'reconstruct' embedding matrices that are not quantized. In order to avoid this, you specifically have to read the embeddings as `Embeddings<VocabWrap, QuantizedArray>` or `Embeddings<VocabWrap, MmapQuantizedArray>`. You can then convert them into `Embeddings<VocabWrap, StorageWrap>` in order to be able to represent both in-memory and mmap'ed arrays. But then at least the command will fail when trying to apply reconstruction on non-quantized embredings.

Reply:
Ah thanks for pointing it out! There is an issue here, in ```finalfusion-rust```, the conversion between one type of ```Embeddings``` to another is implemented through ```impl_embeddings_from```, and currently we don't have specialization to convert embeddings with ```MmapQuantizedArray``` to embeddings with ```StorageWrap```. Should I just do a PR for it first?